integrationtests/testground/main.go

```diff
@@ -198,39 +201,40 @@ func testRound(env *runtime.RunEnv, ic *run.InitContext, round string, desiredAd
 		}
 		msg("Joined thread")
 
+		ic.SyncClient.MustSignalAndWait(ctx, sync.State("ready-"+round+"-phase1"), livePeers)
 		start := time.Now()
-		doneStep1 := sync.State("done-" + round + "-step1")
+		donePhase1 := sync.State("done-" + round + "-phase1")
 		go func() {
-			receivedRecords = thr.WaitForRecords(ctx, expectedRecords)
-			env.R().RecordPoint("round-"+round+"-step-1-elapsed-seconds", time.Since(start).Seconds())
-			msg("Peer #%d done %s network step 1", ic.GlobalSeq, round)
-			ic.SyncClient.MustSignalAndWait(ctx, doneStep1, livePeers)
+			_ = thr.WaitForRecords(ctx, recordsToReceive)
+			env.R().RecordPoint("round-"+round+"-phase-1-elapsed-seconds", time.Since(start).Seconds())
+			msg("Peer #%d done %s network phase 1, received %d records", ic.GlobalSeq, round, recordsToReceive)
+			ic.SyncClient.MustSignalAndWait(ctx, donePhase1, livePeers)
 		}()
 
-		for i, prev := 0, cid.Undef; i < myRecords; i++ {
-			rec, err := thr.AddRecord(ctx, prev)
-			if err != nil {
-				return err
-			}
-			prev = rec.Cid()
+		if err := thr.CreateRecords(ctx, recordsToSend); err != nil {
+			return err
 		}
-		msg("Peer #%d added %d records", ic.GlobalSeq, myRecords)
+		msg("Peer #%d created %d records", ic.GlobalSeq, recordsToSend)
 		// wait until all live peers get correct results
-		<-ic.SyncClient.MustBarrier(ctx, doneStep1, livePeers).C
+		<-ic.SyncClient.MustBarrier(ctx, donePhase1, livePeers).C
 	}
 
-	myRecords = expectedRecords
 	meLive = 1
+	recordsToSend = env.IntParam("records")
 	if isEarlyStop {
 		meLive = 0
-		myRecords = 0
+		recordsToSend = 0
 	}
-	livePeers = publishAndGetTotal(sync.NewTopic("livePeers-"+round+"-step-2", 0), meLive)
-	expectedRecords = publishAndGetTotal(sync.NewTopic("expectedRecords-"+round+"-step-2", 0), myRecords)
+	livePeers = publishAndGetTotal(sync.NewTopic("livePeers-"+round+"-phase-2", 0), meLive)
+	recordsToReceive = publishAndGetTotal(sync.NewTopic("recordsToReceive-"+round+"-phase-2", 0), recordsToSend) + recordsToReceive
+	chCurrentHead := make(chan cid.Cid, 1)
+	topic = sync.NewTopic("current-head-"+round, cid.Cid{})
```

Comment:
This is to publish the log head so other instances can `GetRecord` and traverse the whole log

Reply:
Is this using the testground tooling to send records vs. letting the peers get and push records over the p2p lib?

Reply:
It's just to publish the CIDs of the logs head so each peer can traverse. Is there another way to get them without subscribing?

Reply:
When a peer writes to their log, it will push the new records to the other peers in thread using [this method](https://github.com/textileio/go-threads/blob/master/net/client.go#L251) on the p2p API. So, after thread setup, each peer should have the thread (either by creating it or adding it via your share thread method). After that point, the sharing should happen automatically.

Reply:
Yeah I get it but even though the records are pushed to other peers, how can the caller of go-threads on other peers access them without knowing the head CID?

Reply:
Oh I see, so you want to be able to walk the records in the test to verify everything was received. 

You could do that with the `net/api` `Client` (which is the user API) using the `GetThread` [method](https://github.com/textileio/go-threads/blob/master/net/api/pb/threadsnet.proto#L128), which returns all thread info, including the head, and then use `GetRecord` [method](https://github.com/textileio/go-threads/blob/master/net/api/pb/threadsnet.proto#L134) to walk back with `Prev`.

Reply:
Aha that's much easier! I noticed the `Logs` field of `thread.Info` before but totally forgot that! Thanks!

Reply:
So I'd imagine code would typical either subscribe the thread to get updates, or periodically call `GetThread` to get a full picture at that point. But then the test code needs to wait long enough for the records to be actually synced to other peers. The wait time is tricky to decide. Maybe I need to keep `GetThread` and check if the heads of the logs are all up-to-date?

Reply:
This test does record the time taken to get all records, so I guess it doesn't hurt to do the periodical polling as you mentioned.

Reply:
Done https://github.com/textileio/go-threads/pull/523/commits/4c9ab518efccaccbdae274bb280f2ae7a9ba5cf8